app/models/concerns/foreman/thread_session.rb

```diff
@@ -47,14 +48,12 @@ def current
         end
 
         def current=(o)
-          unless o.nil? || o.is_a?(self)
-            raise(ArgumentError, "Unable to set current User, expected class '#{self}', got #{o.inspect}")
-          end
-
-          if o.is_a?(User)
-            Rails.logger.info "Current user: #{o.login} (#{o.admin? ? 'administrator' : 'regular user'})"
```

Comment:
The problem is the `User.current` set after logging the info about request/params, and it's cleared before we log the info about the response: see sample at https://github.com/theforeman/foreman/pull/4385#issuecomment-287367867. This means we would need to rely on some random log messages from the view to see what user was running that. Keeping at least a info message, not mentioning the login itself, but at least an info if it's a regular user or administrator. could help